### ISSUE TITLE
native crash :JNI DETECTED ERROR IN APPLICATION: use of deleted weak global reference 0xf0f0f0f0f0f0f0f'

### ISSUE THREAD

**Issue by focuseyes360** [realm/realm-java#7582]
### How frequently does the bug occur?

Seen once

### Description

My project just migrated from android support libs to androidx libs, and i test my app ,then i got this crash . 

java stacktrace:
  at io.realm.internal.NativeObjectReference.nativeCleanUp(Native method)
  at io.realm.internal.NativeObjectReference.cleanup(NativeObjectReference.java:87)
  - locked <0x04fb0f31> (a io.realm.internal.NativeContext)
  at io.realm.internal.FinalizerRunnable.run(FinalizerRunnable.java:38)
  at java.lang.Thread.run(Thread.java:818)

### Stacktrace & log output

```shell
** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Tombstone maker: 'xCrash 2.4.9'
Crash type: 'native'
Start time: '2021-10-26T10:58:51.253+0800'
Crash time: '2021-10-26T11:15:08.284+0800'
App ID: 'com.eqtinfo.qutalk'
App version: '3.0.1.L(30001)'
Rooted: 'No'
API level: '23'
OS version: '6.0'
Kernel version: 'Linux version 3.18.22+ #1 SMP PREEMPT Fri Apr 19 16:12:38 CST 2019 (aarch64)'
ABI list: 'arm64-v8a,armeabi-v7a,armeabi'
Manufacturer: 'Xiaomi'
Brand: 'Xiaomi'
Model: 'Redmi Note 4'
Build fingerprint: 'Xiaomi/nikel/nikel:6.0/MRA58K/V10.2.2.0.MBFCNXM:user/release-keys'
ABI: 'arm64'
pid: 14475, tid: 14650, name: RealmFinalizing  >>> com.eqtinfo.qutalk <<<
signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
Abort message: 'art/runtime/java_vm_ext.cc:440] JNI DETECTED ERROR IN APPLICATION: use of deleted weak global reference 0xf0f0f0f0f0f0f0f'
    x0  0000000000000000  x1  000000000000393a  x2  0000000000000006  x3  0000000000000000
    x4  0000000000000000  x5  0000000000000001  x6  0000000000000000  x7  0000000000000000
    x8  0000000000000083  x9  00000000000349c0  x10 000000000235ed90  x11 000000000235eed0
    x12 0000000000000000  x13 0000007f857d4000  x14 0000000000000000  x15 0000000000000000
    x16 0000007f857c6570  x17 0000007f85757b1c  x18 0000007f857d7f50  x19 0000007f61091500
    x20 0000007f61091440  x21 000000000000000b  x22 0000000000000006  x23 0000007f64e48200
    x24 0000007f3e4749c0  x25 0000000000000000  x26 0000007f81b29000  x27 0000000000000000
    x28 0000007f64e48200  x29 0000007f61090330
    sp  0000007f61090330  lr  0000007f857552b8  pc  0000007f85757b24

backtrace:
    #00 pc 000000000006ab24  /system/lib64/libc.so (tgkill+8)
    #01 pc 00000000000682b4  /system/lib64/libc.so (pthread_kill+68)
    #02 pc 0000000000023ae4  /system/lib64/libc.so (raise+28)
    #03 pc 000000000001e284  /system/lib64/libc.so (abort+60)
    #04 pc 00000000004347c4  /system/lib64/libart.so (_ZN3art7Runtime5AbortEv+324)
    #05 pc 00000000001365b0  /system/lib64/libart.so (_ZN3art10LogMessageD2Ev+3136)
    #06 pc 0000000000311568  /system/lib64/libart.so (_ZN3art9JavaVMExt8JniAbortEPKcS2_+2288)
    #07 pc 0000000000311b34  /system/lib64/libart.so (_ZN3art9JavaVMExt9JniAbortFEPKcS2_z+224)
    #08 pc 000000000045b9d0  /system/lib64/libart.so (_ZNK3art6Thread13DecodeJObjectEP8_jobject+548)
    #09 pc 0000000000148114  /system/lib64/libart.so (_ZN3art11ScopedCheck5CheckERNS_18ScopedObjectAccessEbPKcPNS_12JniValueTypeE.constprop.116+1560)
    #10 pc 0000000000164198  /system/lib64/libart.so (_ZN3art8CheckJNI9DeleteRefEPKcP7_JNIEnvP8_jobjectNS_15IndirectRefKindE+528)
    #11 pc 000000000012dd40  /data/app/com.eqtinfo.qutalk-1/lib/arm64/librealm-jni.so
    #12 pc 00000000000b9f1c  /data/app/com.eqtinfo.qutalk-1/lib/arm64/librealm-jni.so
    #13 pc 00000000000ba2c8  /data/app/com.eqtinfo.qutalk-1/lib/arm64/librealm-jni.so
    #14 pc 00000000000a4fe0  /data/app/com.eqtinfo.qutalk-1/lib/arm64/librealm-jni.so (Java_io_realm_internal_NativeObjectReference_nativeCleanUp+12)
    #15 pc 00000000050bd3f8  /data/app/com.eqtinfo.qutalk-1/oat/arm64/base.odex (void io.realm.internal.NativeObjectReference.nativeCleanUp(long, long)+140)
    #16 pc 00000000007fc3c4  [stack]

build id:
    /system/lib64/libc.so (BuildId: 6822ffc6bfdb000cfba4cd5ea3b45a2d. FileSize: 1041456. LastModified: 2019-04-19T16:17:19.000+0800. MD5: 21acfbd9e0678185f7f351109332ffc0)
    /system/lib64/libart.so (BuildId: 7b54192bed1b649a7620718382079a52. FileSize: 7523160. LastModified: 2019-04-19T16:17:19.000+0800. MD5: 7516b4f9c0836b2eb8ecfb05fdd914f2)
    /data/app/com.eqtinfo.qutalk-1/lib/arm64/librealm-jni.so (BuildId: 5b9dff361dcb1f1322c7035b0e6440cc7cb3beec. FileSize: 4482384. LastModified: 1979-12-31T01:01:02.000+0800. MD5: 13ebfca85baff8cd936a55cf4d02d864)
    /data/app/com.eqtinfo.qutalk-1/oat/arm64/base.odex (BuildId: unknown. FileSize: 146846816. LastModified: 2021-10-26T10:58:18.000+0800)
```

### Can you reproduce the bug?

Not yet

### Version

classpath "io.realm:realm-gradle-plugin:6.0.2"

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

android API level: '23' OS version: '6.0' Kernel version: 'Linux version 3.18.22+ #1 SMP PREEMPT Fri Apr 19 16:12:38 CST 2019 (aarch64)' ABI list: 'arm64-v8a,armeabi-v7a,armeabi' Manufacturer: 'Xiaomi' Brand: 'Xiaomi' Model: 'Redmi Note 4'

### Build environment

Android Studio version: 4.1.2
Android Build Tools version: 29.0.2
Gradle version: 3.6.0

**Comment by focuseyes360** [comment]
but now , I upgrade my realm sdk from v6.0.2 to v7.0.2 ,and then the question above  never appear ,but ,when i input something quickly enough  , i got a crash as follow:

java stacktrace:
  at io.realm.internal.NativeObjectReference.nativeCleanUp(Native method)
  at io.realm.internal.NativeObjectReference.cleanup(NativeObjectReference.java:87)
  at io.realm.internal.FinalizerRunnable.run(FinalizerRunnable.java:38)
  at java.lang.Thread.run(Thread.java:919)

pid: 18844, tid: 19710, name: RealmFinalizing  >>> com.eqtinfo.qutalk <<<
signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xf0f0f0f0f0f0f
    x0  0f0f0f0f0f0f0f0f  x1  000000745968e1e8  x2  0000000000000020  x3  0000007435bb9758
    x4  00000074aaacd1e8  x5  00000074aaacd208  x6  0000007435bb93a0  x7  0000007435bb97b8
    x8  000000745968e1e8  x9  8059257f74504e05  x10 0000000000000004  x11 0000000000000000
    x12 0000007435bb9750  x13 0000007435bb95f8  x14 00000000ffffffff  x15 0000000000000000
    x16 000000758a269758  x17 000000758a1f8e80  x18 00000074238e8000  x19 00000074a84daff0
    x20 000000745968e1e8  x21 000000745968e1e8  x22 0000000013de4e40  x23 00000000138e02b0
    x24 00000074862b55e8  x25 00000074aaede600  x26 0000000000000050  x27 000000746d633020
    x28 000000746d6324b0  x29 000000746d631ec0
    sp  000000746d631e00  lr  0000007486340c64  pc  000000748633853c

backtrace:
    #00 pc 000000000019b53c  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #01 pc 00000000001a3c60  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #02 pc 00000000001a3c1c  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #03 pc 00000000001a34f4  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #04 pc 00000000001a3510  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #05 pc 00000000001ae6b8  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #06 pc 00000000001a64a0  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #07 pc 00000000001a645c  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #08 pc 000000000019b028  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #09 pc 000000000011876c  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #10 pc 000000000016e770  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #11 pc 0000000000119680  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #12 pc 00000000001185fc  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #13 pc 0000000000058f60  /memfd:/jit-cache (deleted)

build id:
    /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so (BuildId: 04926858a1a347390f7b63cd2087113180457b0d. FileSize: 5513560. LastModified: 1981-01-01T01:01:02.000+0800. MD5: 5643f46362b662c378e0e9c898ae85e9)
    /memfd:/jit-cache (deleted) (BuildId: unknown. OPEN error: errno = 2, errmsg = No such file or directory)

stack:
         000000746d631d80  0000000000000000
         000000746d631d88  0000007509c00140  [anon:libc_malloc]
         000000746d631d90  000000746d6324b0
         000000746d631d98  000000746d633020
         000000746d631da0  0000000000000050
         000000746d631da8  00000074aaede600  [anon:libc_malloc]
         000000746d631db0  00000074862b55e8  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
         000000746d631db8  00000074aaaccc00  [anon:libc_malloc]
         000000746d631dc0  00000074aaaccc38  [anon:libc_malloc]
         000000746d631dc8  000000745968e210  [anon:libc_malloc]
         000000746d631dd0  00000074aaaccc00  [anon:libc_malloc]
         000000746d631dd8  00000074f940e408  [anon:libc_malloc]
         000000746d631de0  000000746d631e20
         000000746d631de8  000000758a1e928c  /apex/com.android.runtime/lib64/bionic/libc.so (ifree+676)
         000000746d631df0  00000074862b55e8  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
         000000746d631df8  000000758a26b290  [anon:.bss]
    #00  000000746d631e00  00000074aab184b0  [anon:libc_malloc]
         ........  ........
    #01  000000746d631e00  00000074aab184b0  [anon:libc_malloc]
         000000746d631e08  00000074a84daff0  [anon:libc_malloc]
         000000746d631e10  00000074a84daff0  [anon:libc_malloc]
         000000746d631e18  0000007486340c20  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #02  000000746d631e20  00000074a84daf40  [anon:libc_malloc]
         000000746d631e28  00000074863404f8  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #03  000000746d631e30  00000074aab184b0  [anon:libc_malloc]
         000000746d631e38  000000758a1e95b4  /apex/com.android.runtime/lib64/bionic/libc.so (je_free+124)
         000000746d631e40  00000074a84daf40  [anon:libc_malloc]
         000000746d631e48  0000007486340514  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #04  000000746d631e50  000000742074a400  [anon:libc_malloc]
         000000746d631e58  000000748634b6bc  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #05  000000746d631e60  00000074aaede758  [anon:libc_malloc]
         000000746d631e68  00000074863434a4  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #06  000000746d631e70  00000074a982e6d0  [anon:dalvik-LinearAlloc]
         000000746d631e78  00000074862a7d5c  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so (Java_io_realm_internal_NativeObjectReference_nativeCleanUp)
         000000746d631e80  00000074aaede758  [anon:libc_malloc]
         000000746d631e88  0000007486343460  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #07  000000746d631e90  00000074aaede740  [anon:libc_malloc]
         000000746d631e98  000000748633802c  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #08  000000746d631ea0  00000074aaede6b0  [anon:libc_malloc]
         000000746d631ea8  00000074862b5770  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #09  000000746d631eb0  00000074862a7d5c  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so (Java_io_realm_internal_NativeObjectReference_nativeCleanUp)
         000000746d631eb8  00000074aaede620  [anon:libc_malloc]
         000000746d631ec0  000000746d631ee0
         000000746d631ec8  000000748630b774  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #10  000000746d631ed0  00000074aaede600  [anon:libc_malloc]
         000000746d631ed8  8059257f74504e05
         000000746d631ee0  000000746d631f00
         000000746d631ee8  00000074862b6684  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #11  000000746d631ef0  00000074aaede600  [anon:libc_malloc]
         000000746d631ef8  00000000138e02b0  [anon:dalvik-main space (region space)]
         000000746d631f00  000000746d631f20
         000000746d631f08  00000074862b5600  /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
    #12  000000746d631f10  00000074aab2ac00  [anon:libc_malloc]
         000000746d631f18  00000074aab2ac00  [anon:libc_malloc]
         000000746d631f20  000000746d632400
         000000746d631f28  000000009be15f64  /memfd:/jit-cache (deleted)
    #13  000000746d631f30  00000074a982e6a8  [anon:dalvik-LinearAlloc]
         000000746d631f38  0000000000000000
         000000746d631f40  13487df000000001
         000000746d631f48  0000007500000000  /apex/com.android.runtime/etc/icu/icudt63l.dat
         000000746d631f50  0000000000000000
         000000746d631f58  0000000000000000
         000000746d631f60  0000000000000000
         000000746d631f68  0000000000000000
         000000746d631f70  0000000000000000
         000000746d631f78  0000000000000000
         000000746d631f80  0000000000000000
         000000746d631f88  0000000000000000
         000000746d631f90  00000074aab2ac00  [anon:libc_malloc]
         000000746d631f98  0000000000000000
         000000746d631fa0  00000074a982e6d0  [anon:dalvik-LinearAlloc]
         000000746d631fa8  0000000013de4e40  [anon:dalvik-main space (region space)]

**Comment by clementetb** [comment]
Unrolled stacktrace
```
#00 0x0000000000000000 <unknown>
#0 0x000000000019b53c /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
std::__ndk1::default_delete<realm::ParentNode>::operator()(realm::ParentNode*) const
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/memory:2338:5
std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> >::reset(realm::ParentNode*)
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/memory:2593:0
std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> >::~unique_ptr()
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/memory:2547:0
#1 0x00000000001a3c60 /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
std::__ndk1::allocator<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> > >::destroy(std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> >*)
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/memory:1880:64
void std::__ndk1::allocator_traits<std::__ndk1::allocator<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> > > >::__destroy<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> > >(std::__ndk1::integral_constant<bool, true>, std::__ndk1::allocator<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> > >&, std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> >*)
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/memory:1742:0
void std::__ndk1::allocator_traits<std::__ndk1::allocator<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> > > >::destroy<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> > >(std::__ndk1::allocator<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> > >&, std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> >*)
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/memory:1595:0
std::__ndk1::__vector_base<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> >, std::__ndk1::allocator<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> > > >::__destruct_at_end(std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> >*)
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/vector:426:0
#2 0x00000000001a3c1c /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
std::__ndk1::__vector_base<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> >, std::__ndk1::allocator<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> > > >::~__vector_base()
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/vector:463:9
#3 0x00000000001a34f4 /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
std::__ndk1::vector<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> >, std::__ndk1::allocator<std::__ndk1::unique_ptr<realm::ParentNode, std::__ndk1::default_delete<realm::ParentNode> > > >::~vector()
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/vector:555:5
realm::OrNode::~OrNode()
/mnt/jenkins/workspace/realm_realm-core_release_6.0.17@2/build-android-arm64-v8a-Release/../src/realm/query_engine.hpp:1816:0
#4 0x00000000001a3510 /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
realm::OrNode::~OrNode()
/mnt/jenkins/workspace/realm_realm-core_release_6.0.17@2/build-android-arm64-v8a-Release/../src/realm/query_engine.hpp:1816:7
#5 0x00000000001ae6b8 /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
realm::BoolNode<realm::Equal>::~BoolNode()
/mnt/jenkins/workspace/realm_realm-core_release_6.0.17@2/build-android-arm64-v8a-Release/../src/realm/query_engine.hpp:1170:7
#6 0x00000000001a64a0 /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
realm::QueryGroup::~QueryGroup()
/mnt/jenkins/workspace/realm_realm-core_release_6.0.17@2/build-android-arm64-v8a-Release/../src/realm/query.hpp:61:8
std::__ndk1::allocator<realm::QueryGroup>::destroy(realm::QueryGroup*)
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/memory:1880:0
void std::__ndk1::allocator_traits<std::__ndk1::allocator<realm::QueryGroup> >::__destroy<realm::QueryGroup>(std::__ndk1::integral_constant<bool, true>, std::__ndk1::allocator<realm::QueryGroup>&, realm::QueryGroup*)
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/memory:1742:0
void std::__ndk1::allocator_traits<std::__ndk1::allocator<realm::QueryGroup> >::destroy<realm::QueryGroup>(std::__ndk1::allocator<realm::QueryGroup>&, realm::QueryGroup*)
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/memory:1595:0
std::__ndk1::__vector_base<realm::QueryGroup, std::__ndk1::allocator<realm::QueryGroup> >::__destruct_at_end(realm::QueryGroup*)
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/vector:426:0
#7 0x00000000001a645c /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
std::__ndk1::__vector_base<realm::QueryGroup, std::__ndk1::allocator<realm::QueryGroup> >::~__vector_base()
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/vector:463:9
#8 0x000000000019b028 /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
std::__ndk1::vector<realm::QueryGroup, std::__ndk1::allocator<realm::QueryGroup> >::~vector()
/opt/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/vector:555:5
realm::Query::~Query()
/mnt/jenkins/workspace/realm_realm-core_release_6.0.17@2/build-android-arm64-v8a-Release/../src/realm/query.cpp:152:0
#9 0x000000000011876c /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
realm::ConstTableView::~ConstTableView()
/Users/cm/Realm/realm-java/realm/realm-library/distribution/realm-core/core-5.0.15/include/realm/table_view.hpp:198:5
#10 0x000000000016e770 /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
realm::Results::~Results()
/Users/cm/Realm/realm-java/realm/realm-library/src/main/cpp/object-store/src/results.cpp:33:19
#11 0x0000000000119680 /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
realm::_impl::ObservableCollectionWrapper<realm::Results>::~ObservableCollectionWrapper()
/Users/cm/Realm/realm-java/realm/realm-library/src/main/cpp/observable_collection_wrapper.hpp:44:44
#12 0x00000000001185fc /data/app/com.eqtinfo.qutalk-EzttkW84WI7ndJzwkS5uEA==/lib/arm64/librealm-jni.so
finalize_results(long)
/Users/cm/Realm/realm-java/realm/realm-library/src/main/cpp/io_realm_internal_OsResults.cpp:40:5
#13 0x0000000000058f60 /memfd:/jit-cache (deleted)
```

**Comment by clementetb** [comment]
@focuseyes360 do you got some sample project to help us reproduce the issue?

**Comment by cmelchior** [comment]
Hi @focuseyes360 If this is still an issue, we would recommend upgrading to the latest version: 10.10.1, since we had a number of fixes around memory management merged since version 6.0.2.

If this is still an issue after that, please reopen this issue.
